packages/java-parser/src/productions/classes.js

```diff
@@ -202,9 +202,12 @@ function defineRules($, t) {
   // https://docs.oracle.com/javase/specs/jls/se16/html/jls-8.html#jls-VariableDeclaratorList
   $.RULE("variableDeclaratorList", () => {
     $.SUBRULE($.variableDeclarator);
-    $.MANY(() => {
-      $.CONSUME(t.Comma);
-      $.SUBRULE2($.variableDeclarator);
+    $.MANY({
+      GATE: () => !tokenMatcher(this.LA(3).tokenType, t.Identifier),
```

Comment:
It took me some time to understand why it could cause a problem. Could we add add a comment here in order to make it clear why we need this (to distinguish from patternList) ? And maybe a test sample in the parser's tests ?

For instance:
```ts
it ("should parse pattern list", () => {
  const input = `A a, B b`;
  expect(() => javaParser.parse(input, "patternList")).to.not.throw();
});
```

Reply:
Done!